bionic/flow.py

```diff
@@ -115,12 +115,84 @@ def install_provider(self, provider):
 
         return self._set_provider(provider).touch()
 
-    def add_case(self, name, case_key, value):
-        provider = self.get_provider(name).copy_if_mutable()
+    def group_names_together(self, names):
+        providers = [self.get_provider(name) for name in names]
+        providers_set = set(providers)
+
+        # If there's only one provider, then the grouping is already done.
+        if len(providers_set) == 1:
+            provider = providers[0]
+            if not set(provider.attrs.names) == set(names):
```

Comment:
any specific reason to not use `!=` instead?

Reply:
Nope, just an oversight -- thanks!